contents/docs/libraries/js/index.mdx

```diff
@@ -456,6 +456,153 @@ const handleCookieConsent = (consent) => {
+
+> **Note:** amending and rejecting events should be done with care. It can cause unexpected results in parts of PostHog.
+
+### Not all events are processed by `beforeSend`.
+
+Events which are to do with billing or where sampling/editing would break the PostHog feature are never processed by `beforeSend`
+
+* $web_experiment_applied
+* $$client_ingestion_warning
+* $feature_enrollment_update
+* $feature_flag_called
+* survey dismissed
+* survey sent
+* survey shown
+* $snapshot
+* $identify
+* $groupidentify
+* $create_alias
+
+### Redacting information in events
+
+`beforeSend` gives you one place to edit or redact information before it is sent to PostHog.
+
+#### Redact URLs in event properties
+
+```
+posthog.init('my_token', {
+    beforeSend: (event): event | null => {
+        const redactedProperties = {}
+        for (const [key, value] of Object.entries(event.properties || {})) {
+            if (key.includes(url)) {
+                redactedProperties[key] = value.replace(/customer_id/\d+/, 'customer_id/1234567')
+            } else {
+                redactedProperties[key] = value
+            }
+        }
+        event.properties = redactedProperties
+        
+        return event
+    }
+})
+```
+
+#### Redact person properties in $set or $set_once
+
+```
+posthog.init('my_token', {
+    beforeSend: (event): event | null => {
+        event.$set = {
+            ...event.$set,
+            homeAddress: '******'
+        }
+        event.$set_once = {
+            ...event.$set,
+            // TODO should these two account for initial / current prefixes?
+            homeAddress: '******'
+        }
+
+        return event
+    }
+})
+```
+
+#### Redact URLs in heatmap data
+
+```
+posthog.init('my_token', {
+    beforeSend: (event): event | null => {
+        if (event.event !== '$heatmap') {
+            const redactedHeatmapData = {}
+            
+            for(const [url, data] of Object.entries(event.properties.$heatmap_data)) {
+                redactedHeatmapData[url.replace(/private_id\/\d+/, 'private_id/1234567')] = data
+            }
+            event.properties.$heatmap_data = redactedHeatmapData
+        }
+        
+        return event
+    }
+})
+```
+
+### Sampling events
+
+Sampling lets you choose to send only a percentage of events to PostHog. It is a good way to control your costs without having to completely turn off features of the SDK.
+
+Some functions of PostHog - for example much of web analytics - relies on receiving all events. Sampling $pageview or $pageleave events in particular can cause unexpected results.
+
+#### Sampling events using our provided customization
+
+If you're using NPM we offer a pre-built function to sample by event name. 
+You can import this or copy it from our repo.
+
+```
+posthog.init('my_token', {
+    // capture only half of dead click and web vitals events
+    beforeCapture: sampleByEvent(['$dead_click', '$web_vitals'], 50)
```

Comment:
@simfish85 I don't know if you want to user test this approach with $secret_customer that were asking about web vitals sampling?